src/mono/mono/metadata/custom-attrs.c

```diff
@@ -389,11 +391,11 @@ load_cattr_value (MonoImage *image, MonoType *t, MonoObject **out_obj, const cha
 		return val;
 	}
 	case MONO_TYPE_VALUETYPE:
-		if (m_class_is_enumtype (t->data.klass)) {
-			type = mono_class_enum_basetype_internal (t->data.klass)->type;
+		if (m_class_is_enumtype (m_type_data_get_klass (t))) {
```

Comment:
Could we at least put m_type_data_get_klass (t) into a local only doing call once for this block?